Backend/app/provider-platform/dynamic-offer-driver-app/Main/src/Domain/Action/Dashboard/Management/NammaTag/Handle.hs

```diff
@@ -2,18 +2,46 @@ module Domain.Action.Dashboard.Management.NammaTag.Handle (kaalChakraHandle) whe
 
 import qualified Domain.Types.Person as DPerson
 import EulerHS.Prelude hiding (id)
+import qualified Kernel.External.Types
 import Kernel.Types.Id
 import Kernel.Utils.Common
+import qualified Lib.Scheduler.JobStorageType.SchedulerType as QAllJ
 import qualified Lib.Yudhishthira.Event.KaalChakra as KaalChakra
 import qualified Lib.Yudhishthira.Types
+import SharedLogic.Allocator
+import Storage.Beam.SchedulerJob ()
 import qualified Storage.Queries.Person as QPerson
 
+type HandlerFlow m r = (EsqDBFlow m r, MonadFlow m, CacheFlow m r, HasField "jobInfoMap" r (Map Text Bool), HasField "schedulerSetName" r Text, HasField "schedulerType" r Kernel.External.Types.SchedulerType, HasField "maxShards" r Int)
+
+createFetchUserDataJob :: HandlerFlow m r => Lib.Yudhishthira.Types.UpdateKaalBasedTagsJobReq -> UTCTime -> m ()
+createFetchUserDataJob req scheduledTime = do
+  let jobData = Lib.Yudhishthira.Types.mkKaalChakraJobDataFromUpdateTagData req True
+  maxShards <- asks (.maxShards)
+  case req.chakra of
+    Lib.Yudhishthira.Types.Daily -> QAllJ.createJobByTime @_ @'Daily scheduledTime maxShards jobData
+    Lib.Yudhishthira.Types.Weekly -> QAllJ.createJobByTime @_ @'Weekly scheduledTime maxShards jobData
+    Lib.Yudhishthira.Types.Monthly -> QAllJ.createJobByTime @_ @'Monthly scheduledTime maxShards jobData
+    Lib.Yudhishthira.Types.Quarterly -> QAllJ.createJobByTime @_ @'Quarterly scheduledTime maxShards jobData
+
+createUpdateUserData :: HandlerFlow m r => Lib.Yudhishthira.Types.RunKaalChakraJobReq -> Id Lib.Yudhishthira.Types.Event -> UTCTime -> m ()
+createUpdateUserData req eventId scheduledTime = do
```

Comment:
this function creates and updates user tags, not user data right?

Reply:
yeah....will rename